libs/spirit-islander/data-access/src/lib/app-state.service.ts

```diff
@@ -32,30 +32,47 @@ interface AppState {
 export class AppStateService {
   private readonly _oldKey = 'CONFIG_NEW';
   private readonly _key = 'CONFIG';
-  private _config: Config = this._getConfig();
+  private _config: Config = this._getConfig(); // TODO: consider initializing _state in the constructor so this _config doesn't have to be a long-lived class member
   private _state = new State<AppState>({
-    page: Page.Config,
     config: this._config,
     validCombos: undefined,
     gameSetup: createGameSetup(this._config, getValidCombos(this._config)),
   });
 
   state$ = this._state.get().pipe(tap(({ config }) => this._setConfig(config)));
 
-  constructor(private _localStorageService: LocalStorageService) {}
+  constructor(
+    private _localStorageService: LocalStorageService,
+    private _router: Router
+  ) {}
 
-  edit(): void {
-    this._state.updateProp('page', Page.Config);
+  navigateToGameSetup(config: Config): void {
+    this._router.navigate(['game-setup'], {
+      queryParams: mapConfigToQueryParams(config),
+    });
   }
 
-  generate(config: Config, validCombos: Combo[]): void {
-    this._state.updateProp('config', config);
-    this._state.updateProp('validCombos', validCombos);
-    this._state.updateProp('gameSetup', createGameSetup(config, validCombos));
-    this._state.updateProp('page', Page.GameSetup);
+  navigateToConfig(): void {
+    this._router.navigate(['config']);
   }
 
-  regenerate(): void {
+  processParams(params: ParamMap): void {
+    try {
+      const config = mapParamMapToConfig(params);
+      const validCombos = getValidCombos(config);
+
+      this._state.updateProp('config', config);
+      this._state.updateProp('validCombos', getValidCombos(config));
+      this._state.updateProp('gameSetup', createGameSetup(config, validCombos));
```

Comment:
File a ticket to investigate if `createGameSetup` should only accept the `config` and then internally derive the `validCombos`. Calculating and passing in `validCombos` every time is annoying

Reply:
https://github.com/johnnycopes/atocha/issues/280